lib/badge.js

```diff
@@ -0,0 +1,33 @@
+export const CLEAR_BADGE_ACTION = 'CLEAR_BADGE'
+
+export const dispatchClearBadgeAction = () => navigator.serviceWorker?.controller?.postMessage({ action: CLEAR_BADGE_ACTION })
+
+export const badgingApiSupported = (sw = window) => 'setAppBadge' in sw.navigator
+
+export const permissionGranted = async (sw = window, name = 'notifications') => {
```

Comment:
Nitpick: You're not using these functions anywhere else. I would then not export them

Reply:
Good call, I can fix that. I probably used them in a previous commit but ended up refactoring quite a bit.